galsim/sed.py

```diff
@@ -47,78 +68,113 @@ class SED(object):
+    the case-insensitive aliases 'nm', 'nanometer', 'nanometers', 'A', 'Ang', 'Angstrom', or
+    'Angstroms'.  Note that `astropy.units.spectral` includes not only units with dimensions of
+    length, but also frequency, energy, or wavenumber.
+
+    The return value of `spec` should be a spectral density with units specified by `flux_type`,
+    which should be an instance of `astropy.units.Unit` of equivalency class
+    `astropy.units.spectral_density`, or one of the case-insensitive aliases:
+        1. 'flambda':  erg/wave_type/cm^2/s, where wave_type is as above.
+        2. 'fnu':      erg/Hz/cm^2/s
+        3. 'fphotons': photons/wave_type/cm^2/s, where wave_type is as above.
+        4. '1':        dimensionless
+    Note that the `astropy.units.spectral_density` class includes units with dimensions of
+    [energy/time/area/unit-wavelength], [energy/time/area/unit-frequency],
+    [photons/time/area/unit-wavelength], and so on.
+
+    Finally, the optional `fast` keyword option is used to specify when unit and dimension changes
+    are executed, particularly for SEDs specified by a LookupTable.  If `fast=True`, the default,
+    then the input units/dimensions may be converted to an internal working unit before
+    interpolation in wavelength is performed.  Alternatively, `fast=False` implies that
+    interpolation should take place in the native units of the input `spec`, and subsequently flux
+    density converted to photons/cm^2/s/nm afterwards.  Generally, the former option is faster, but
+    may be less accurate since interpolation and dimensionality conversion do not commute.  One
+    consequence of using `fast=True` is that __call__ can not accept `astropy.units.Quantity`s in
+    this case.
 
     @param spec          Function defining the z=0 spectrum at each wavelength.  See above for
                          valid options for this parameter.
     @param wave_type     String specifying units for wavelength input to `spec`.
     @param flux_type     String specifying what type of spectral density `spec` represents.  See
                          above for valid options for this parameter.
     @param redshift      Optionally shift the spectrum to the given redshift. [default: 0]
+    @param fast          Convert units on initialization instead of on __call__. [default: True]
     """
-    def __init__(self, spec, wave_type, flux_type, redshift=0.,
-                 _blue_limit=None, _red_limit=None, _wave_list=None, _spec=None ):
+    def __init__(self, spec, wave_type, flux_type, redshift=0., fast=True,
+                 _blue_limit=None, _red_limit=None, _wave_list=None,
+                 _spectral=None, _dimensionless=None):
 
         self._orig_spec = spec  # Save this for pickling
-        self._spec = _spec      # This is orig_spec turned into a function
-
+        # If these are known at construction time, use them.
+        self._spectral = _spectral
+        self._dimensionless = _dimensionless
```

Comment:
This was originally because I needed to check for both in the case something compatible with neither was passed as flux_type.  I think your solution below though simplifies this.  I did keep dimensionless as a convenience as suggested.